jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/select/DisableNativeSelectTests.java

```diff
@@ -0,0 +1,62 @@
+package io.github.epam.angular.tests.elements.complex.select;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+
+import static io.github.com.pages.sections.SelectSection.disableNativeSelect;
+
+public class DisableNativeSelectTests extends TestsSelectBase {
+    private static final String DISABLED = "disabled";
+
+    @BeforeMethod(alwaysRun = true)
+    public void before() {
+        disableNativeSelect.show();
+    }
+
+    @Test
+    public void checkLabelValue() {
+        disableNativeSelect.label().has().value("Choose an option");
+    }
+
+    @Test
+    public void verifyCheckboxCanDisableSelect() {
+        pickDisableSelectCheckboxAsChecked();
+        disableNativeSelect.waitFor().assertThat().hasAttribute(DISABLED);
+    }
+
+    @Test
+    public void checkEnabledOptionCanBeSelectedByIndex() {
+        pickDisableSelectCheckboxAsUnchecked();
+        if (disableNativeSelect.base().get().getAttribute(DISABLED) == null) {
+            disableNativeSelect.select(2);
+            disableNativeSelect.is().selected(VOLVO);
+        }
+    }
+
+    @Test
+    public void checkDisabledOptionCannotBeSelectedByName() {
+        pickDisableSelectCheckboxAsUnchecked();
+        if (disableNativeSelect.base().get().getAttribute(DISABLED) == null) {
+            String preselectedValue = disableNativeSelect.selected();
+            disableNativeSelect.select(SAAB);
+            disableNativeSelect.is().selected(preselectedValue);
+        }
+    }
+

```

Comment:
It seems like an “if” condition is not a good choice for a test. In the case of “false”, we simply skip the test and do not know if the test is working or not.

Reply:
I agree. Fixed.